### PR TITLE
Tracking events for icon library

### PR DESCRIPTION
**Pull request by zamoore**
### :pushpin: Summary

If merged, this PR will add tracking events to the icon library page.

### :hammer_and_wrench: Detailed description

- When a user searches (debounced by 1s), we track `Icon Library - Search - SEARCH_QUERY`
- When a user selects icon size, we track `Icon Library - Size Selector - SELECTED_SIZE`
- When a user selects how they want to group icons, we track `Icon Library - Group by Selector - GROUP_BY`

### :link: External links

https://hashicorp.atlassian.net/browse/HDS-3382

### CODE REVIEW

**Review by Dhaulagiri — APPROVED** [review]
Nice on! Let's keep an eye on the events next week to see if the icon search gets too noisy. (I can show you how to see them if we didn't talk about that already)

File: website/app/modifiers/doc-track-event.js [hunk 1]
```diff
@@ -9,14 +9,10 @@ import { inject as service } from '@ember/service';
...
+  @service eventTracking;
 
   modify(element, _positional, named) {
-    if (
-      // Only attempt to do something if we are in the right environment
-      this.fastboot.isFastBoot ||
-      window.fathom == null
-    ) {
+    if (!this.eventTracking.isEnabled) {
       // comment this line if you want to test in your local environment
```
Comment (Dhaulagiri) [inline website/app/modifiers/doc-track-event.js:16]: [nit] I think this was added via suggestion in a previous PR, but trying this out I don't see the events actually working.

Comment (didoo) [reply website/app/modifiers/doc-track-event.js:16]: 🤔

File: website/app/modifiers/doc-track-event.js [hunk 2]
```diff
@@ -9,14 +9,10 @@ import { inject as service } from '@ember/service';
...
-  @service fastboot;
+  @service eventTracking;
 
   modify(element, _positional, named) {
-    if (
-      // Only attempt to do something if we are in the right environment
-      this.fastboot.isFastBoot ||
-      window.fathom == null
-    ) {
+    if (!this.eventTracking.isEnabled) {
```
Comment (Dhaulagiri) [inline website/app/modifiers/doc-track-event.js:15]: Looking at this with fresher eyes I'm not sure this entire `if` block is actually necessary as the same check now happens [here](https://github.com/hashicorp/design-system/pull/2167/files#diff-b5adddb0d69ebd2ee22d67a269141428dda6e318176c851ef8158f9fd700cbeeR18) as well

File: website/app/modifiers/doc-track-event.js (hunk 1 above)
Comment (Dhaulagiri) [reply website/app/modifiers/doc-track-event.js:16]: Are you adding something to actually log the string or how are you seeing the event?

Comment (zamoore) [reply website/app/modifiers/doc-track-event.js:16]: AFAIK you would need to comment it out as well as adding a log.

File: website/app/modifiers/doc-track-event.js (hunk 2 above)
Comment (zamoore) [reply website/app/modifiers/doc-track-event.js:15]: It's not strictly necessary, but it prevents the event listener from being added unnecessarily

File: website/app/modifiers/doc-track-event.js (hunk 1 above)
Comment (didoo) [reply website/app/modifiers/doc-track-event.js:16]: Yes, as @zamoore says, I'm commenting out the call to Fathom and using a simple `console.log` instead.

File: website/app/modifiers/doc-track-event.js (hunk 2 above)
Comment (didoo) [reply website/app/modifiers/doc-track-event.js:15]: +1 on @zamoore approach

File: website/app/modifiers/doc-track-event.js (hunk 1 above)
Comment (Dhaulagiri) [reply website/app/modifiers/doc-track-event.js:16]: Ok. I think I was reading this too literally then as the _only_ thing you needed to do.

**Pull request merged**